### DISCUSSION TITLE
ROS 2 Humble in Docker on Pi5 can see Galactic Create3 topics - Do I need Humble Firmware?

### DISCUSSION THREAD

**Discussion by slowrunner** [iRobotEducation/create3_docs#488] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

ROS 2 Humble Docker on Pi5 PiOS Bookworm

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

G.5.3

### Which RMW is your robot running?

CycloneDDS

### Is the robot connected to a network? If so, what is the network type?

Yes: 2.4GHz Home WiFi 

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

Yes

### What is the Adapter Board's USB/BLE Toggle currently switched to?

Bluetooth (default)

### Describe your question.

Just out of curiosity, I started up the Docker ROS 2 Humble on my Raspberry Pi5 mounted on top of my new Create3-Wali robot, knowing the Create3 is running Galactic and CycloneDDS.  

Somehow the Galactic Create3 topics are fully visible in the Humble container.  RMW_IMPLEMENTATION is not set in the container, so I have no idea how it is working.

I tried sending an action goal, but my Docker does not have irobot-create-msgs installed yet.

Do I need to upgrade the Create3 firmware to Humble?

(I was planning on rebuilding the container with ROS 2 Galactic, but perhaps not needed?)

This is probably a bad idea since any question I ask would be suspect - I think I should upgrade the firmware to Humble.

**Comment by alsora** [comment, marked as answer]
ROS 2 doesn't guarantee communication between different RMW implementations or different distributions.
Some things will work, but I strongly recommend not mixing them.

Moreover, note that the default RMW implementation in humble is fast-dds, so you are not only mixing distribution, but also middleware.
From what I remember, actions will likely not work in this setup.
But again, please update the robot to Humble (and make sure that robot and pi use the same middleware.

We won't be able to help you debug issues if there are inconsistencies between distributions or middlewares.
